doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -476,13 +495,33 @@ Requests to the environment
       Use the :cmd:`Add @table` and :cmd:`Remove @table` commands to update the set of
       blacklisted strings.
 
-.. cmd:: Locate @qualid
+.. cmd:: Locate @locatable
+
+  .. insertprodn locatable locatable
+
+  .. prodn::
+     locatable ::= @smart_qualid
+     | Term @smart_qualid
+     | Module @qualid
+     | Ltac @qualid
+     | Library @qualid
+     | File @string
 
    This command displays the full name of objects whose name is a prefix
-   of the qualified identifier :n:`@qualid`, and consequently the |Coq| module in
+   of the specified value, and consequently the |Coq| module in
    which they are defined. It searches for objects from the different
    qualified namespaces of |Coq|: terms, modules, Ltac, etc.
 
+   The various :n:`locatable`\s have the following behavior:
+
+   -  :n:`@smart_qualid` - displays the full name of objects whose name is a prefix
+      of :n:`@smart_qualid`, thereby showing the module they are defined in.
+   -  :n:`Term @smart_qualid` - limits the search to terms
+   -  :n:`Module @qualid` - limits the search to modules
+   -  :n:`Library @qualid` - limits the search to libraries
+   -  :n:`Ltac @qualid` - limits the search to Ltac-defines tactics
+   -  :n:`File @string` - ???
```

Comment:
Same here: it wouldn't necessarily hurt to repeat `Locate`; and then it might help to use something else than `:n:`, to indicate that this is the defining location for each of these commands.  Otherwise the string `Locate Module` doesn't appear anywhere in the file, which doesn't help with looking stuff up.

Reply:
Searching for "Locate" takes them right to this info.  No need for "Locate Module".

Reply:
In this case, I'm not quite sure I agree. I feel like these `Locate Module` are distinctive prefixes much like `Print Module`.

Reply:
Note that the index can list all the "Locate *" by adding a `:name:`, similar to we did for `Axiom` et al.  Not sure if that responds to your thought.